v21.1/developer-guide-overview.md

```diff
@@ -4,32 +4,65 @@ summary: An overview of common tasks that come up when you build an application
 
-## Common Tasks
+## Guide contents
 
-- [Install a Postgres Client](install-client-drivers.html)
-- [Connect to the Database](connect-to-the-database.html)
-- [Insert Data](insert-data.html)
-- [Query Data](query-data.html)
-- [Update Data](update-data.html)
-- [Delete Data](delete-data.html)
-- [Run Multi-Statement Transactions](run-multi-statement-transactions.html)
-- [Error Handling and Troubleshooting](error-handling-and-troubleshooting.html)
-- [Make Queries Fast](make-queries-fast.html)
+- [Overview](#cockroachdb-basics)
+- [Connect to a Cluster](install-client-drivers.html)
+- [Design a Schema](schema-design-overview.html)
+- [Write Data](insert-data.html)
+- [Read Data](query-data.html)
+- [Debug Your Application](debug-and-error-logs.html)
+- [Optimize Performance](make-queries-fast.html)
 
-## Sample Apps and Tutorials
+## CockroachDB basics
 
-For simple but complete sample apps and tutorials, see [Hello World](hello-world-example-apps.html).
+Before you start using CockroachDB, you should understand a couple important mechanics of the database:
 
-For more advanced sample apps and tutorials, see [Build a Spring App with CockroachDB](build-a-spring-app-with-cockroachdb-jdbc.html) and [Develop and Deploy a Multi-Region Web Application](multi-region-overview.html).
+- [How transactions work in CockroachDB](#how-transactions-work-in-cockroachdb)
+- [How applications interact with CockroachDB](#how-applications-interact-with-cockroachdb)
 
-{% include cockroach_u_pydev.md %}
+Note that the sections that follow were written for the purposes of orienting application developers. For more detailed technical documentation on CockroachDB's architecture, see our [architecture documentation](architecture/overview.html).
+
+### How transactions work in CockroachDB
+
+CockroachDB is designed to make your data scalable and consistent. All stored records are replicated across a distributed deployment of database instances, and all database transactions committed in CockroachDB satisfy [ACID properties](https://en.wikipedia.org/wiki/ACID) and the [CAP theorem](https://en.wikipedia.org/wiki/CAP_theorem). This means that when you store data in CockroachDB, the data will be valid and anomaly-free, even in the event of a system error or power failure.
+
+To guarantee that database operations are [atomic](https://en.wikipedia.org/wiki/Atomicity_(database_systems)) (the "A" of the [ACID properties](https://en.wikipedia.org/wiki/ACID)), CockroachDB executes all statements in the context of an atomic [database transaction](https://en.wikipedia.org/wiki/Database_transaction). If a transaction succeeds, all data mutations are applied together with virtual simultaneity. If any part of a transaction fails, the entire transaction is aborted, and the database is left unchanged.
```

Comment:
maybe i'd say something like "all data mutations are applied together simultaneously from the perspective of operations outside of the transaction"